### PR TITLE
Julia: add mentoring notes for robot-name

### PR DESCRIPTION
**Pull request by cmcaine**

### CODE REVIEW

**Comment by cmcaine** [comment]
@SaschaMann

File: tracks/julia/exercises/robot-name/mentoring.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+The main point of this exercise is to learn about mutable structs and to reinforce the idea of writing methods for your own types.
+But there's also some trickyness in the problem, too.
+
+Julia things to get right
+
+- fields should have explicit types (structs with `Any` fields are slower, ref Julia performance guide)
+
+Solutions should ideally give an error if they run out of names (instead of looping forever or issuing duplicate names).
+
+There are broadly three ways of solving this (in ascending goodness):
```
Comment (SaschaMann) [inline tracks/julia/exercises/robot-name/mentoring.md:12]:
```suggestion
There are broadly four ways of solving this (in ascending goodness):
```

File: tracks/julia/exercises/robot-name/mentoring.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
+## Mentoring notes
+
+The main point of this exercise is to learn about mutable structs and to reinforce the idea of writing methods for your own types.
+But there's also some trickyness in the problem, too.
+
+Julia things to get right
+
+- fields should have explicit types (structs with `Any` fields are slower, ref Julia performance guide)
```
Comment (SaschaMann) [inline tracks/julia/exercises/robot-name/mentoring.md:8]:
```suggestion
- fields should have explicit types (structs with `Any` fields are slower, ref [Julia performance guide](https://docs.julialang.org/en/v1/manual/performance-tips/index.html))
```

**Comment by cmcaine** [comment]
I don't know how your review got dismissed :/

Thanks for your comments, I've included them and this is now good to go :)

**Comment by SaschaMann** [comment]
> I don't know how your review got dismissed :/

Me neither ¯\_(ツ)_/¯

**Pull request merged**